rflx/pyrflx/typevalue.py

```diff
@@ -224,39 +240,594 @@ class OpaqueValue(TypeValue):
     def __init__(self, vtype: Opaque) -> None:
         super().__init__(vtype)
 
-    def assign(self, value: bytes, check: bool = True) -> None:
+    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
         self._value = value
 
-    def assign_bitvalue(self, value: str, check: bool = True) -> None:
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        self._value = bytes(value)
 
-        while len(value) % 8 != 0:
-            value = "0" + value
+    @property
+    def length(self) -> int:
+        self._raise_initialized()
+        return len(self._value) * 8
 
-        bytestring = b"".join(
-            [int(value[i : i + 8], 2).to_bytes(1, "big") for i in range(0, len(value), 8)]
-        )
+    @property
+    def value(self) -> bytes:
+        self._raise_initialized()
+        return self._value
+
+    @property
+    def to_bytes(self) -> bytes:
+        self._raise_initialized()
+        return self._value
+
+    @property
+    def to_bitstring(self) -> Bitstring:
+        self._raise_initialized()
+        return Bitstring(format(int.from_bytes(self._value, "big"), f"0{self.length}b"))
+
+    @property
+    def accepted_type(self) -> type:
+        return bytes
+
+    @property
+    def literals(self) -> Mapping[Name, Expr]:
+        return {}
+
+
+class ArrayValue(TypeValue):
+
+    _value: List[TypeValue]
+    _element_list_type: Type
+    _is_message_array: bool
+    _all_messages: List[model.Message]
```

Comment:
It is usually not needed to declare types for attributes in this way, as mypy is able to infer the attribute types in the constructor.